packaging/os/rhn_register.py

```diff
@@ -223,17 +223,17 @@ def register(self, enable_eus=False, activationkey=None, profilename=None):
             Register system to RHN.  If enable_eus=True, extended update
             support will be requested.
         '''
-        register_cmd = "/usr/sbin/rhnreg_ks --username='%s' --password='%s' --force" % (self.username, self.password)
+        register_cmd = ['/usr/sbin/rhnreg_ks', '--username', self.username, '--password', self.password, '--force']
         if self.module.params.get('server_url', None):
-            register_cmd += " --serverUrl=%s" % self.module.params.get('server_url')
+            register_cmd.extend('--serverUrl', self.module.params.get('server_url'))
```

Comment:
needs_revision
@mscherer extend takes one argument, a list of additional items rather than separate items. See https://docs.python.org/2.6/tutorial/datastructures.html for details.